extensions/ql-vscode/src/contextual/templateProvider.ts

```diff
@@ -45,14 +45,22 @@ export class TemplateQueryDefinitionProvider implements vscode.DefinitionProvide
   }
 
   private async getDefinitions(uriString: string): Promise<vscode.LocationLink[]> {
-    return getLocationsForUriString(
-      this.cli,
-      this.qs,
-      this.dbm,
-      uriString,
-      KeyType.DefinitionQuery,
-      (src, _dest) => src === uriString
-    );
+    return await withProgress({
```

Comment:
Here and elsewhere: This `return await` can just be a `return` right? Otherwise we'll await the outcome of the promise, then wrap up the result in a second promise anyway, and return that.

Reply:
It's redundant. I can remove. Though, I don't think the behaviour is any different due to hoisting of promises. It is essentially this snippet:

```js
async function a() {
  throw new Error('1')
}

async function b1() {
  return await a()
}

async function b2() {
  return a()
}

async function c() {
  try {
    await b1()
  } catch (e) {
    console.log(e)
  }
  try {
    await b2()
  } catch (e) {
    console.log(e)
  }
}

c()
```

Both errors are properly captured.